lib/quicly.c

```diff
@@ -4909,7 +4909,10 @@ void quicly_amend_ptls_context(ptls_context_t *ptls)
 int quicly_encrypt_address_token(void (*random_bytes)(void *, size_t), ptls_aead_context_t *aead, ptls_buffer_t *buf,
                                  size_t start_off, const quicly_address_token_plaintext_t *plaintext)
 {
-    int ret;
+    int ret = 0;
+
+    if (start_off < sizeof(quicly_address_token_plaintext_t))
+	goto Exit; 
```

Comment:
I am not sure if this change is necessary. `start_off` does not have any relationship with `quicly_address_token_plaintext_t`.

As stated in the doc-comment of the [doc-comment of the function declaration](https://github.com/h2o/quicly/blob/master/include/quicly.h#L950-L955), the role of the function is to append token to the buffer being provided, considering bytes between start_off and `buf->off` as AAD.

Also, the function is expected to return an error code (0 means success). I'd assume that this is an error case?